clarity/src/vm/ast/parser_v2/mod.rs

```diff
@@ -0,0 +1,3227 @@
+                }
+                let trait_name = match ClarityName::try_from(name.clone()) {
+                    Ok(id) => id,
+                    Err(_) => {
+                        self.add_diagnostic(
+                            ParseErrors::IllegalTraitName(name.clone()),
+                            trait_span.clone(),
+                        )?;
+                        let mut placeholder = PreSymbolicExpression::placeholder(format!(
+                            "'{}.{}",
+                            contract_id, name,
+                        ));
+                        placeholder.span = span;
+                        return Ok(placeholder);
+                    }
+                };
+                let trait_id = TraitIdentifier {
+                    name: trait_name,
+                    contract_identifier: contract_id,
+                };
+                let mut expr = PreSymbolicExpression::field_identifier(trait_id);
+                expr.span = span;
+                Ok(expr)
+            } else {
+                let contract_principal = PrincipalData::Contract(contract_id);
+                let mut expr =
+                    PreSymbolicExpression::atom_value(Value::Principal(contract_principal));
+                expr.span = span;
+                Ok(expr)
+            }
+        } else {
+            let mut expr = PreSymbolicExpression::atom_value(Value::Principal(
+                PrincipalData::Standard(principal),
+            ));
+            expr.span = span;
+            Ok(expr)
+        }
+    }
+
+    fn read_sugared_principal(&mut self, mut span: Span) -> ParseResult<PreSymbolicExpression> {
+        let (name, contract_span) = match self.next_token() {
+            Some(PlacedToken {
+                span: contract_span,
+                token: Token::Ident(ident),
+            }) => {
+                span.end_line = contract_span.end_line;
+                span.end_column = contract_span.end_column;
+                (ident, contract_span)
+            }
+            Some(PlacedToken {
+                span: token_span,
+                token,
+            }) => {
+                self.add_diagnostic(ParseErrors::ExpectedContractIdentifier, token_span.clone())?;
+                let mut placeholder =
+                    PreSymbolicExpression::placeholder(format!(".{}", token.reproduce()));
+                span.end_line = token_span.end_line;
+                span.end_column = token_span.end_column;
+                placeholder.span = span;
+                return Ok(placeholder);
+            }
+            None => {
+                self.add_diagnostic(ParseErrors::ExpectedContractIdentifier, span.clone())?;
+                let mut placeholder = PreSymbolicExpression::placeholder(".".to_string());
+                placeholder.span = span;
+                return Ok(placeholder);
+            }
+        };
+
+        let contract_name = match ContractName::try_from(name.clone()) {
```

Comment:
Should this code be checking that the contract name's length isn't too long here?  This isn't done in `ContractName::try_from()`.

Reply:
Fixed and tested in 9d2634ef9